archlinuxcn/llvm-git/PKGBUILD

```diff
@@ -0,0 +1,421 @@
+         '295c343dcd457dc534662f011d7cff1a')
+sha512sums=('SKIP'
+            '75e743dea28b280943b3cc7f8bbb871b57d110a7f2b9da2e6845c1c36bf170dd883fca54e463f5f49e0c3effe07fbd0db0f8cf5a12a2469d3f792af21a73fcdd')
+options=('staticlibs' 'debug')
+
+# NINJAFLAGS is an env var used to pass commandline options to ninja
+# NOTE: It's your responbility to validate the value of $NINJAFLAGS. If unsure, don't set it.
+
+_python2_optimize() {
+    python2 -m compileall "$@"
+    python2 -O -m compileall "$@"
+}
+
+_python3_optimize() {
+    python -m compileall "$@"
+    python -O -m compileall "$@"
+    python -OO -m compileall "$@"
+}
+
+pkgver() {
+    cd llvm-project/llvm
+
+    # This will almost match the output of `llvm-config --version` when the
+    # LLVM_APPEND_VC_REV cmake flag is turned on. The only difference is
+    # dash being replaced with underscore because of Pacman requirements.
+    local _pkgver=$(awk -F 'MAJOR |MINOR |PATCH |)' \
+                        'BEGIN { ORS="." ; i=0 } \
+             /set\(LLVM_VERSION_/ { print $2 ; i++ ; if (i==2) ORS="" } \
+             END { print "\n" }' \
+                        CMakeLists.txt)_r$(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+    echo "$_pkgver"
+
+}
+
+prepare() {
+    cd llvm-project
+    # llvm-project contains a lot of stuff, remove parts that aren't used by this package
+    rm -rf debuginfo-tests libclc libcxx libcxxabi llgo openmp parallel-libs pstl libc
+
+    rm -rf "$srcdir"/fakeinstall*
+}
+
+build() {
+    CFLAGS+=' -I/usr/include/tensorflow'
+    CXXFLAGS+=' -I/usr/include/tensorflow'
+    CFLAGS+=' -fasynchronous-unwind-tables'
+    CXXFLAGS+=' -fasynchronous-unwind-tables'
+    LDFLAGS+=' -fasynchronous-unwind-tables'
+
+    #  export CC="clang -v"
+    #  export CXX="clang++ -v"
+
+    cmake \
+        -B _build64 \
+        -S llvm-project/llvm  -G Ninja \
+        -D CMAKE_C_FLAGS="$CFLAGS" \
+        -D CMAKE_CXX_FLAGS="$CXXFLAGS" \
+        -D CMAKE_BUILD_TYPE=Release \
+        -D CMAKE_INSTALL_PREFIX=/usr \
+        -D LLVM_APPEND_VC_REV=ON \
+        -D LLVM_HOST_TRIPLE="$CHOST" \
+        -D LLVM_ENABLE_RTTI=ON \
+        -D LLVM_ENABLE_FFI=ON \
+        -D LLVM_BINUTILS_INCDIR=/usr/include \
+        -D FFI_INCLUDE_DIR="$(pkg-config --variable=includedir libffi )" \
+        -D LLVM_BUILD_LLVM_DYLIB=ON \
+        -D LLVM_LINK_LLVM_DYLIB=ON \
+        -D LLVM_INSTALL_UTILS=ON \
+        -D LLVM_BUILD_DOCS=ON \
+        -D LLVM_INCLUDE_DOCS=ON \
+        -D LLVM_ENABLE_DOXYGEN=OFF \
+        -D CLANG_INCLUDE_DOCS=ON \
+        -D LLVM_EXPERIMENTAL_TARGETS_TO_BUILD=AVR \
+        -D LLVM_ENABLE_SPHINX=ON \
+        -D SPHINX_OUTPUT_HTML=OFF \
+        -D SPHINX_WARNINGS_AS_ERRORS=OFF \
+        -D POLLY_ENABLE_GPGPU_CODEGEN=ON \
+        -D LLVM_VERSION_SUFFIX="" \
+        -D LLDB_USE_SYSTEM_SIX=1 \
+        -D LLVM_ENABLE_PROJECTS="lldb;polly;compiler-rt;lld;clang-tools-extra;clang" \
+        -D TENSORFLOW_C_LIB_PATH="/usr/include/tensorflow/"
```

Comment:
I'm doing a test compilation right now so I might be able to answer this question myself in a few hours, but does this create a runtime dependency on tensorflow, or is that purely compile time?

Reply:
It's also a bit strange that you are passing a include path to something called "lib_path"...